### PR TITLE
Hotfix 2.28.4 (Part I): Small fixes for some ACRA crashes & signature widget

### PR DESCRIPTION
**Pull request by phillipm**
2.28 versions of the following PRs:

- https://github.com/dimagi/commcare-android/pull/1405
- https://github.com/dimagi/commcare-android/pull/1403
- https://github.com/dimagi/commcare-android/pull/1408

### CODE REVIEW

File: app/src/org/commcare/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -1172,8 +1172,14 @@ void displayMessage(String message, boolean bad, boolean suppressToast) {
     }
 
     public static boolean isDemoUser() {
-        User u = CommCareApplication._().getSession().getLoggedInUser();
-        return (User.TYPE_DEMO.equals(u.getUserType()));
+        try {
+            User u = CommCareApplication._().getSession().getLoggedInUser();
+            return (User.TYPE_DEMO.equals(u.getUserType()));
+        } catch (SessionUnavailableException e) {
+            // Default to a normal user: this should only happen if session
```
Comment (wpride) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: Shouldn't we redirect to login in this case? Is there the possibility a Demo user could end up with some functionality they shouldn't and end up crashing?

**Comment by wpride** [comment]
one question then gtm

File: app/src/org/commcare/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (phillipm) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: Redirection to login should happen in `onResume` of any activity that extends `SessionAwareCommCareActivity`. Adding this `try/catch` ensures we don't crash if `isDemoUser` is invoked somewhere before `onResume` gets called.

Comment (phillipm) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: But you're right, it is good to be cautious about this default value! I just checked and the uses of `isDemoeUser` toggle warning dialogs and enable the settings menu. If those don't or do show up when they shouldn't we should be fine.

**Pull request merged**